Libraries/acdriver/Sources/CompileAction.cpp

```diff
@@ -27,10 +62,303 @@ CompileAction::
 {
 }
 
+static bool
+CompileAsset(
+    std::shared_ptr<xcassets::Asset::Catalog> const &catalog,
+    Options const &options,
+    CompileOutput *compileOutput,
+    Result *result,
+    std::shared_ptr<xcassets::Asset::Asset> const &parent,
+    std::shared_ptr<xcassets::Asset::Asset> const &asset);
+
+template<typename T>
+static bool
+CompileChildren(
+    std::shared_ptr<xcassets::Asset::Catalog> const &catalog,
+    Options const &options,
+    CompileOutput *compileOutput,
+    Result *result,
+    std::shared_ptr<xcassets::Asset::Asset> const &parent,
+    std::vector<T> const &assets)
+{
+    bool success = true;
+
+    for (auto const &asset : assets) {
+        if (!CompileAsset(catalog, options, compileOutput, result, parent, asset)) {
+            success = false;
```

Comment:
Are you purposefully trying to compile each asset even if one of them fails?

You could also do:

`success &= CompileAsset(...)`